Mage.Sets/src/mage/cards/v/Vault101BirthdayParty.java

```diff
@@ -0,0 +1,119 @@
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCardInLibrary;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ *
+ * @author The_REAL_Boudidi
+ */
+public final class Vault101BirthdayParty extends CardImpl {
+
+    public Vault101BirthdayParty(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{W}");
+        
+        this.subtype.add(SubType.SAGA);
+
+        // (As this Saga enters and after your draw step, add a lore counter. Sacrifice after III.)
+        SagaAbility sagaAbility = new SagaAbility(this);
+
+        // I -- Create a 1/1 white Human Soldier creature token and a Food token.
+        sagaAbility.addChapterEffect(this, SagaChapter.CHAPTER_I,
+                new CreateTokenEffect(new HumanWarriorToken()),
+                new CreateTokenEffect(new FoodToken()));
+
+        // II, III -- You may put an Aura or Equipment card from your hand or graveyard onto the battlefield.
+        // If an Equipment is put onto the battlefield this way, you may attach it to a creature you control.
+        sagaAbility.addChapterEffect(this, SagaChapter.CHAPTER_II, SagaChapter.CHAPTER_III,
+                new Vault101BirthdayPartyEffect());
+
+        this.addAbility(sagaAbility);
+
+    }
+
+    private Vault101BirthdayParty(final Vault101BirthdayParty card) {
+        super(card);
+    }
+
+    @Override
+    public Vault101BirthdayParty copy() {
+        return new Vault101BirthdayParty(this);
+    }
+
+}
+
+class Vault101BirthdayPartyEffect extends OneShotEffect {
+
+    private static final FilterCard filter = new FilterCard("");
+
+    static {
+        filter.add(Predicates.or(
+                SubType.AURA.getPredicate(),
+                SubType.EQUIPMENT.getPredicate()
+        ));
+    }
+
+    public Vault101BirthdayPartyEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "put an Aura or Equipment card from your hand or graveyard onto the battlefield. "
+                         + "If an Equipment is put onto the battlefield this way, you may attach it to a creature you control." ;
+    }
+
+    private Vault101BirthdayPartyEffect(final mage.cards.v.Vault101BirthdayPartyEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public mage.cards.v.Vault101BirthdayPartyEffect copy() {
+        return new mage.cards.v.Vault101BirthdayPartyEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        Cards cards = new CardsImpl();
+        cards.addAllCards(player.getHand().getCards(filter, game));
+        cards.addAllCards(player.getGraveyard().getCards(filter, game));
+        TargetCard targetCard = new TargetCard(Zone.ALL, filter);

```

Comment:
this needs to be `withNotTarget(true)` as well since the card text doesn't use the word "target"

Reply:
Use `new TargetCard(0, 1, Zone.ALL, filter).withNotTarget(true);`

the min num targets of 0 is an adequate implementation for "you may"

Reply:
Thanks, I made the change.

Reply:
did you get the "you may" working? You didn't add the min num targets 0, and I don't see any other code that would account for that

Reply:
It's done now, I got a little mixed up with my commits.  But I set the minimum targets to 0 and the "you may" now works correctly.  Thanks a lot for your help.